cli/ssh_internal_test.go

```diff
@@ -0,0 +1,58 @@
+package cli
+
+import (
+	"net/url"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/codersdk"
+)
+
+const (
+	fakeOwnerName     = "fake-owner-name"
+	fakeServerURL     = "https://fake-foo-url"
+	fakeWorkspaceName = "fake-workspace-name"
+)
+
+func TestVerifyWorkspaceOutdated(t *testing.T) {
+	t.Parallel()
+
+	serverURL, err := url.Parse(fakeServerURL)
+	require.NoError(t, err)
+
+	client := codersdk.Client{URL: serverURL}
+
+	t.Run("Up-to-date", func(t *testing.T) {
```

Comment:
Sorry, I didn't look too closely at these tests on my first pass. I should point out that the preferred way to test in the `coder/coder` repo is write external tests (`package cli_test`).

With these types of tests, we'd do a [bit more setup](https://github.com/coder/coder/blob/a5cc1970cf4799b67b28133323ea7adc648e8d84/cli/ssh_test.go#L69-L102) for the test so that we can then test the output of running `coder ssh ...`.

(Not a blocker this PR, but just a heads up for future PRs.)

Reply:
Ok, I see now. I can use existing coderdtest APIs. I need to create a workspace to simulate a template update (`CreateTemplateVersion`). Something like `setupWorkspaceForAgent`, but customized.

I can adjust the implementation, but not sure if the internal_test isn't just simpler. Do you have any preference?